src/Microsoft.ML.Data/DataView/CompositeRowToRowMapper.cs

```diff
@@ -70,7 +75,11 @@ public Row GetRow(Row input, Func<int, bool> active)
             var deps = new Func<int, bool>[InnerMappers.Length];
             deps[deps.Length - 1] = active;
             for (int i = deps.Length - 1; i >= 1; --i)
-                deps[i - 1] = InnerMappers[i].GetDependencies(deps[i]);
+            {
+                var outputColumns = InnerMappers[i].OutputSchema.Where(c => deps[i](c.Index));
+                var cols = InnerMappers[i].GetDependencies(outputColumns).ToArray();
+                deps[i - 1] = c => cols.Count() > 0 ? cols.Any(col => col.Index == c) : false;
```

Comment:
>Count() [](start = 40, length = 7)

`Length`. Note that `Count()` the method will actually iterate and literally count.